crates/holochain_persistence_api/src/eav/eavi.rs

```diff
@@ -103,12 +107,12 @@ pub type Index = i64;
 // type Source ...
 /// The basic struct for EntityAttributeValue triple, implemented as AddressableContent
 /// including the necessary serialization inherited.
-#[derive(PartialEq, Eq, Hash, Clone, Debug, Serialize, Deserialize)]
+#[derive(PartialEq, Eq, Hash, Clone, Debug, Serialize, Deserialize, PartialOrd, Ord)]
 pub struct EntityAttributeValueIndex<A: Attribute> {
+    index: Index,
```

Comment:
Did you change this field ordering so that `index` is the first deciding factor for comparison/equals?

Reply:
Yes the problem was that when checking for duplicates some places would use composite keys but only collect into a BTreeSet which uses only the index as the key